sdk/go/common/resource/plugin/debugging.go

```diff
@@ -14,7 +14,12 @@
 
 package plugin
 
+import "github.com/pulumi/pulumi/sdk/v3/go/common/workspace"
+
 type DebugEventEmitter interface {
+	// Indicates whether debugging is enabled for the given plugin.
+	ShouldDebugPlugin(workspace.PluginSpec) bool
```

Comment:
The spec here never seems to actually be used?

Reply:
I intend this to enable fine-grained configuration in the future, for example `--attach-debugger="name=node-components".  With the default/initial behavior being "attach to the program and to source-based plugins".